.github/workflows/build-and-test.yaml

```diff
@@ -0,0 +1,28 @@
+name: build and test
+
+on:
+  push:
+    branches: [production, staging]
+  pull_request:
+    branches: [production, staging]
+
+jobs:
+  build:
+    name: build-test
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
+          cache: yarn
+      - name: build
+        run: |
+          export CI=false
```

Comment:
Why CI=false?

Reply:
When `CI=true`, warnings are treated as errors. The build produces many warnings, so the app will only compile when `CI=false`.  I opened [a ticket](https://sagebionetworks.jira.com/browse/DHP-1113) to clean up the warnings, so we can set `CI=true`.